### ISSUE TITLE
Installation: Seurat depends indirectly on Bioconductor package multtest

### ISSUE THREAD

**Issue by HenrikBengtsson** [satijalab/seurat#2675]
FYI, in R 3.6.2 (on Linux), trying to `install.packages("Seurat")` on a fresh account, results in:
```r
ERROR: dependency ‘metap’ is not available for package ‘Seurat’
* removing ‘/wynton/home/bengtsson/hb-test/R/x86_64-pc-linux-gnu-library/3.6-CBI/Seurat’

The downloaded source packages are in
	‘/tmp/RtmpHOCsiQ/downloaded_packages’
Warning messages:
1: In install.packages("Seurat") :
  installation of package ‘mutoss’ had non-zero exit status
2: In install.packages("Seurat") :
  installation of package ‘metap’ had non-zero exit status
3: In install.packages("Seurat") :
  installation of package ‘Seurat’ had non-zero exit status
```

This is because [mutoss](https://cran.r-project.org/web/packages/mutoss/) depends on Bioconductor package 'multtest'.  Because of this, the installation instructions need to be preceded by:

```r
install.packages("BiocManager")
BiocManager::install("multtest")
```

After this, 
```r
install.packages("Seurat")
```
works.

**Comment by andrewwbutler** [comment]
We had noticed that this was potentially causing problems for people. There was some discussion on various forums ([here](https://bioinformatics.stackexchange.com/questions/3365/r-package-development-how-does-one-automatically-install-bioconductor-packages) as one example) that recommended adding `biocViews:` to the `DESCRIPTION` file but that only seems to work if installing via devtools. Do you happen to know if there's a best practice for handling this case (CRAN package that depends directly or indirectly on bioconductor packages)? 

I suppose one method would be to move these dependencies to "Suggests" and just prompt the user to install if missing when they hit a relevant function.

**Comment by HenrikBengtsson** [comment]
> Do you happen to know if there's a best practice for handling this case (CRAN package that depends directly or indirectly on bioconductor packages)?

I don't but my best guess is that it's only about documentation right now.

FWIW: CRAN uses the term "mainstream repositories" to refer to both CRAN and Bioconductor, i.e. Bioconductor is a recognized repos.  The packages can specify their own `Additional_repositories` in the DESCRIPTION file.  The latter was introduced a few years ago and allow users to `Suggets:` on packages in non-maintstream repositories and help `R CMD check` validate these.  Point is, CRAN put in a piece of machinery to support and recognize non-CRAN packages.  When this was introduced, I asked Kurt Hurnik (CRAN Team) if it was on the road map to automatically install from non-CRAN packages too.  The answer was "yes, ideally, but when that could/will happen is unknown".  So, if/when they start autoinstalling non-CRAN packages, I'd expect Bioconductor to be the first to be supported.   But until then, I think textual instructions is the best we can do.

Bioconductor is a bit tricky because its repos depends on curring version of R and time of the year, Bioc release or Bioc devel.


> I suppose one method would be to move these dependencies to "Suggests" and just prompt the user to install if missing when they hit a relevant function.

Yes, I've taken that strategy for some of my packages where the dependency was not that important, i.e. it was only used by a few as an optional feature.

**Comment by andrewwbutler** [comment]
I see. It definitely seems like a tricky problem. Was hoping there might be a work around but I appreciate the insight! 

We've updated the install instructions on the website to hopefully mitigate some of the confusion for the current release. I think we'll go with the "Suggests" approach for the next release.

**Comment by andrewwbutler** [comment]
Suggests strategy implemented as of a7ac8fc908e740b3570fc0a25047f173f3bd619e on the `develop` branch.
